datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -598,19 +662,39 @@ mod tests {
             ),
             vec![1]
         );
+
```

Comment:
I don't think this new test case covers the new code in this PR (as `col IS NULL` doesn't result in a literal guarantee). Is the idea to extend the test coverage?

Reply:
If I remove the check `if has_null && value.is_null() { known_not_present = false; break; }`, the unit test of `row_group_pruning_predicate_eq_null_expr` would fail.

Reply:
I filed https://github.com/apache/arrow-datafusion/issues/8688 to track simplifying expressions that have null literals in them (e.g. `X IN (NULL)`)